Detections/AuditLogs/SuspiciousServicePrincipalcreationactivity.yaml

```diff
@@ -0,0 +1,99 @@
+id: 6852d9da-8015-4b95-8ecf-d9572ee0395d
+name: Suspicious Service Principal creation activity
+description: |
+  'This alert will detect creation of an SPN, permissions granted, credentials cretaed, activity and deletion of the SPN in a time frame (default 10 minutes)'
+severity: Low
+requiredDataConnectors:
+  - connectorId: AzureActiveDirectory
+    dataTypes:
+      - AuditLogs
+      - AADServicePrincipalSignInLogs
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - CredentialAccess
+  - PrivilegeEscalation
+  - InitialAccess
+relevantTechniques:
+query: |
+  let timeframe = 10m;
+let lookback = 1d;
+let account_created =
+AuditLogs 
+| where ActivityDisplayName == "Add application"
+| where Result == "success"
+| extend AppID = tostring(AdditionalDetails[1].value)
+| extend creationTime = ActivityDateTime;
+let account_deleted =
+AuditLogs 
+| where OperationName == "Remove service principal"
+| where Result == "success"
+| extend AppID = tostring(AdditionalDetails[1].value)
+| extend deletionTime = ActivityDateTime;
+account_created | join kind= inner (account_deleted) on AppID, AppID
+| where deletionTime - creationTime < lookback
+| where tolong(deletionTime - creationTime) >= 0
+
+let timeframe = 60m;
```

Comment:
One reason KQL validation is failing because you have multiple query blocks, you can only have one.  Also, make sure you have at least a one space indent for the query block.  Check your YAML structure before submitting with a validator such as https://codebeautify.org/yaml-validator/

Reply:
My mistake! 

Query Reviewed!!!

Thanks